velox/exec/tests/SortBufferTest.cpp

```diff
@@ -0,0 +1,356 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/exec/SortBuffer.h"
+#include <gtest/gtest.h>
+#include <type/Type.h>
+#include <filesystem>
+#include "velox/common/file/FileSystems.h"
+#include "velox/exec/tests/utils/OperatorTestBase.h"
+#include "velox/exec/tests/utils/TempDirectoryPath.h"
+#include "velox/vector/fuzzer/VectorFuzzer.h"
+#include "velox/vector/tests/utils/VectorTestBase.h"
+
+using namespace facebook::velox::exec;
+using namespace facebook::velox::exec::test;
+using namespace facebook::velox;
+using namespace facebook::velox::memory;
+
+namespace {
+RowVectorPtr buildExpectedSortResult(std::vector<RowVectorPtr> inputVectors) {}
+
+void verifyData(
+    RowVectorPtr output,
+    RowVectorPtr expectedResult,
+    int expectedResultIndex) {
+  ASSERT_GT(output->size(), 0);
+  ASSERT_GE(expectedResult->size(), output->size());
+  for (int i = 0; i < output->size(); ++i) {
+    for (int index = 0; index < output->childrenSize(); index++) {
+      ASSERT_TRUE(output->childAt(i)->equalValueAt(
+          expectedResult->childAt(expectedResultIndex)->wrappedVector(),
+          index,
+          index));
+    }
+  }
+}
+} // namespace
+
+namespace facebook::velox::functions::test {
+
+class SortBufferTest : public OperatorTestBase {
+ public:
+  const RowTypePtr inputType_ = ROW(
+      {{"c0", BIGINT()},
+       {"c1", INTEGER()},
+       {"c2", SMALLINT()},
+       {"c3", REAL()},
+       {"c4", DOUBLE()},
+       {"c5", VARCHAR()}});
+  // Specifies the sort columns ["c4", "c1"].
+  std::vector<column_index_t> sortColumnIndices_{4, 1};
+  std::vector<CompareFlags> sortCompareFlags_{
+      {true, true, false, false},
+      {true, true, false, false}};
+
+  const int64_t kMaxBytes = 20LL << 20; // 20 MB
+  const std::shared_ptr<memory::MemoryPool> rootPool_{
+      memory::defaultMemoryManager().addRootPool("SortBufferTest", kMaxBytes)};
+  const std::shared_ptr<memory::MemoryPool> pool_{
+      rootPool_->addLeafChild("SortBufferTest", kMaxBytes)};
```

Comment:
20MB is too small for a memory pool? Set it to 500MB by default and you can create a custom memory pool with small capacity for testing purpose.